### PR TITLE
#78998 - create delete support for endpoints in EventController

### PR DESCRIPTION
**Pull request by mszymeshop**
### What
Adds two DELETE actions for endpoints. They are separated as optional parameters are not supported by swagger and autorest.

### CODE REVIEW

File: src/CaptainHook.Api/Controllers/EventsController.cs [hunk 1]
```diff
@@ -69,6 +69,59 @@ public async Task<IActionResult> PutWebhook([FromRoute] string eventName, [FromR
...
+                    EndpointNotFoundInSubscriberError endpointNotFound => BadRequest(endpointNotFound),
+                    EntityNotFoundError entityNotFoundError => NotFound(entityNotFoundError),
+                    DirectorServiceIsBusyError directorServiceIsBusyError => Conflict(directorServiceIsBusyError),
+                    ReaderDeleteError readerDeleteError => UnprocessableEntity(readerDeleteError),
+                    ReaderDoesNotExistError readerDoesNotExistError => UnprocessableEntity(readerDoesNotExistError),
+                    CannotUpdateEntityError cannotUpdateEntityError => StatusCode(StatusCodes.Status500InternalServerError, cannotUpdateEntityError),
+                    _ => StatusCode(StatusCodes.Status500InternalServerError, error)
+                },
+                subscriberDto => Created($"/{eventName}/subscriber/{subscriberName}/webhooks/endpoint/{selector}", subscriberDto)
+            );
```
Comment (tabman83) [inline src/CaptainHook.Api/Controllers/EventsController.cs:122]: Is this correct?

Comment (mszymeshop) [reply src/CaptainHook.Api/Controllers/EventsController.cs:122]: What exactly? I think I am missing the meaning of the comment.

Comment (tabman83) [reply src/CaptainHook.Api/Controllers/EventsController.cs:122]: Taking it offline

Comment (mszymeshop) [reply src/CaptainHook.Api/Controllers/EventsController.cs]: Thanks Nino, I have now changed this to Ok as this is in sync with requirements.

**Comment by dawidkeshop** [comment]
But we should reuse these error -> HTTP response mappings soon, probably once we unify errors response format.

**Pull request closed without merging**